core/modules/wmgr/python/dbMgr.py

```diff
@@ -0,0 +1,692 @@
+    response = json.jsonify(result=_tblDict(dbName, tblName))
+    response.status_code = 201
+    return response
+
+
+@dbService.route('/<dbName>/tables/<tblName>', methods=['DELETE'])
+def tableDelete(dbName, tblName):
+    """
+    Drop a table and optionally all chunk/overlap tables.
+
+    Following parameters are expected to come in a request (in query string):
+        dropChunks: boolean flag, false by default, accepted values:
+                    '0', '1', 'yes', 'no', 'false', 'true'
+    """
+
+    _log.debug('request: %s', request)
+    _log.debug('tableDelete() DELETE => drop table')
+
+    # validate names
+    _validateDbName(dbName)
+    _validateTableName(tblName)
+
+    # get options
+    dropChunks = _getArgFlag(request.args, 'dropChunks', False)
+    _log.debug('tableDelete(): dropChunks: %s', dropChunks)
+
+    dbConn = Config.instance().dbConn()
+    if not dbConn.dbExists(dbName):
+        raise ExceptionResponse(404, "DatabaseMissing", "Database %s does not exist" % dbName)
+
+    try:
+        # drop chunks first
+        nchunks = 0
+        if dropChunks:
+            # regexp matching all chunk table names
+            tblre = re.compile('^' + tblName + '(FullOverlap)?_[0-9]+$')
+            for table in dbConn.listTables(dbName):
+                if tblre.match(table):
+                    _log.debug('tableDelete(): dropping chunk table %s.%s', dbName, table)
+                    dbConn.dropTable(table, dbName)
+                    nchunks += 1
+
+        # drop main table
+        _log.debug('tableDelete(): dropping main table %s.%s', dbName, tblName)
+        dbConn.dropTable(tblName, dbName)
+
+    except db.DbException as exc:
+        _log.error('tableCreate(): Exception when dropping tables: %s', exc)
+        if exc.errCode() == db.DbException.TB_DOES_NOT_EXIST:
+            chunkMsg = ""
+            if nchunks:
+                chunkMsg = ", but {0} chunk tables have been dropped".format(nchunks)
+            raise ExceptionResponse(404, "TableMissing",
+                                    "Table %s.%s does not exist%s" % (dbName, tblName, chunkMsg))
+        raise
+
+    return json.jsonify(result=_tblDict(dbName, tblName))
+
+
+@dbService.route('/<dbName>/tables/<tblName>/chunks', methods=['GET'])
+def chunks(dbName, tblName):
+    """ Return the list of chunks in a table. For non-chunked table empty list is returned. """
+
+    _log.debug('request: %s', request)
+    _log.debug('chunks() GET => get chunk list')
+
+    # validate params
+    _validateDbName(dbName)
+    _validateTableName(tblName)
+
+    # check the db exists (listTables() does not fail on non-existing databases)
+    dbConn = Config.instance().dbConn()
+    if not dbConn.tableExists(tblName, dbName):
+        raise ExceptionResponse(404, "TableMissing", "Table %s.%s does not exist" % (dbName, tblName))
+
+    # regexp matching chunk table names
+    # TODO: we need some central location for things like this
```

Comment:
No, this is not xrootd interface, but mysql interface (it answers completely different question).